tests/ui/traits/next-solver/gat-wf.rs

```diff
@@ -0,0 +1,14 @@
+// Make sure that, like the old trait solver, we end up requiring that the WC of
```

Comment:
This is also the cause for the failures in `tests/ui/traits/const-traits/predicate-entailment-passes.rs`. Where clauses must be equal, which kinda makes the GATs in that test useless.

Reply:
Kinda prefer an explicit test. 91833 is a mess. I'll add flag, tho, whoops.